packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -2,20 +2,917 @@
+    if (_screenSize != updatedScreenSize) {
+      _screenSize = updatedScreenSize;
+      setState(() {
+        final Rect anchorRect = widget.getRect() ?? _viewRect;
+        final BoxConstraints constraints = widget.viewConstraints ?? widget.viewTheme.constraints ?? widget.viewDefaults.constraints!;
+        final Size updatedViewSize = Size(math.max(constraints.minWidth, anchorRect.width), _viewRect.height);
+        switch (Directionality.of(context)) {
+          case TextDirection.ltr:
+            final Offset updatedPosition = anchorRect.topLeft;
+            _viewRect = updatedPosition & updatedViewSize;
+            break;
+          case TextDirection.rtl:
+            final Offset topLeft = Offset(math.max(anchorRect.right - updatedViewSize.width, 0.0), anchorRect.top);
+            _viewRect = topLeft & updatedViewSize;
+        }
+      });
+    }
+  }
+
+  Widget viewBuilder(Iterable<Widget> suggestions) {
+    if (widget.viewBuilder == null) {
+      return MediaQuery.removePadding(
+        context: context,
+        removeTop: true,
+        child: ListView(
+          children: suggestions.toList()
+        ),
+      );
+    }
+    return widget.viewBuilder!(suggestions);
+  }
+
+  void updateSuggestions() {
+    setState(() {
+      result = widget.suggestionsBuilder(context, _controller);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget defaultLeading = IconButton(
+      icon: const Icon(Icons.arrow_back),
+      onPressed: () { Navigator.of(context).pop(); },
+      style: const ButtonStyle(tapTargetSize: MaterialTapTargetSize.shrinkWrap),
+    );
+
+    final List<Widget> defaultTrailing = <Widget>[
+      IconButton(
+        icon: const Icon(Icons.close),
+        onPressed: () {
+          _controller.clear();
+          updateSuggestions();
+        },
+      ),
+    ];
+
+    final Color effectiveBackgroundColor = widget.viewBackgroundColor
+      ?? widget.viewTheme.backgroundColor
+      ?? widget.viewDefaults.backgroundColor!;
+    final Color effectiveSurfaceTint = widget.viewSurfaceTintColor
+      ?? widget.viewTheme.surfaceTintColor
+      ?? widget.viewDefaults.surfaceTintColor!;
+    final double effectiveElevation = widget.viewElevation
+      ?? widget.viewTheme.elevation
+      ?? widget.viewDefaults.elevation!;
+    final BorderSide? effectiveSide = widget.viewSide
+      ?? widget.viewTheme.side
+      ?? widget.viewDefaults.side;
+    OutlinedBorder effectiveShape = widget.viewShape
+      ?? widget.viewTheme.shape
+      ?? widget.viewDefaults.shape!;
+    if (effectiveSide != null) {
+      effectiveShape = effectiveShape.copyWith(side: effectiveSide);
+    }
+    final Color effectiveDividerColor = widget.dividerColor
+      ?? widget.viewTheme.dividerColor
+      ?? widget.dividerTheme.color
+      ?? widget.viewDefaults.dividerColor!;
+    final MaterialStateProperty<TextStyle?>? effectiveTextStyle = widget.viewHeaderTextStyle
```

Comment:
If these styles really need to be MaterialStateProperties then they can't be merged like this.

Reply:
Thanks for catching this. Change the type to `TextStyle?`